sql/connect/client/jdbc/src/main/scala/org/apache/spark/sql/connect/client/jdbc/SparkConnectStatement.scala

```diff
@@ -86,7 +88,10 @@ class SparkConnectStatement(conn: SparkConnectConnection) extends Statement {
     operationId = null
     resultSet = null
 
-    val df = conn.spark.sql(sql)
+    var df = conn.spark.sql(sql)
+    if (limitRow > 0) {
+      df = df.limit(limitRow)
```

Comment:
It seems that the issue with (SELECT * FROM table LIMIT 3).limit(5) still persists. Although Spark will optimize it.